### PR TITLE
[MNOE-447] Add admin dashboard templates:

### PR DESCRIPTION
**Pull request by hedudelgado**
- Add controller
- Add index and destroy
- Add tests
- Update routes
- Add views

### CODE REVIEW

File: api/app/controllers/mno_enterprise/jpi/v1/admin/dashboard_templates_controller.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  class Jpi::V1::Admin::DashboardTemplatesController < Jpi::V1::Admin::BaseResourceController
+
+    #==================================================================
+    # Instance methods
+    #==================================================================
+    # GET /mnoe/jpi/v1/admin/dashboard_templates
+    def index
+      if params[:terms]
+        # For search mode
+        @dashboard_templates = []
```
Comment (x4d3) [inline api/app/controllers/mno_enterprise/jpi/v1/admin/dashboard_templates_controller.rb]: You should not user current_user.dashboard, but MnoEnterprise::Dashboard directly, and filter on the dashboard where dashboard_type = 'template'

Comment (x4d3) [inline api/app/controllers/mno_enterprise/jpi/v1/admin/dashboard_templates_controller.rb]: You should add a scope in MnoEnterprise::Dashboard, which would be `-> where(dashboard_type: 'template')`

File: api/app/controllers/mno_enterprise/jpi/v1/admin/dashboard_templates_controller.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        head status: :ok
+      else
+        render_bad_request('destroy dashboard template', 'Unable to destroy dashboard template')
+      end
+    end
+
+    private
+
+    def dashboard_templates
+      @dashboard_templates ||= current_user.dashboards
```
Comment (x4d3) [inline api/app/controllers/mno_enterprise/jpi/v1/admin/dashboard_templates_controller.rb]: `MnoEnterprise::Dashboard.templates`

File: api/app/controllers/mno_enterprise/jpi/v1/admin/dashboard_templates_controller.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    end
+
+    def dashboard_template
+      @dashboard_template ||= current_user.dashboards.find(params[:id].to_i)
```
Comment (x4d3) [inline api/app/controllers/mno_enterprise/jpi/v1/admin/dashboard_templates_controller.rb]: `MnoEnterprise::Dashboard.find(params[:id].to_i)`

File: api/app/controllers/mno_enterprise/jpi/v1/admin/dashboard_templates_controller.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+    # Allows all metadata attrs to be permitted, and maps it to :settings
+    # for the Her "meta_data" issue.
+    def dashboard_template_params
+      params.require(:dashboard_template).permit(*whitelisted_params).tap do |whitelisted|
+        whitelisted[:settings] = params[:dashboard_template][:metadata] || {}
+      end
+      .except(:metadata)
+    end
+    alias :dashboard_template_update_params  :dashboard_template_params
```
Comment (x4d3) [inline api/app/controllers/mno_enterprise/jpi/v1/admin/dashboard_templates_controller.rb]: why do yo uneed these aliases ?

File: api/app/controllers/mno_enterprise/jpi/v1/admin/dashboard_templates_controller.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    # GET /mnoe/jpi/v1/admin/dashboard_templates/1
+    def show
+      dashboard_template
+      render_not_found('dashboard template') unless @dashboard_template
+    end
+
+    # POST /mnoe/jpi/v1/admin/dashboard_templates
+    def create
+      if @dashboard_template = dashboard_templates.create(dashboard_template_create_params)
```
Comment (x4d3) [inline api/app/controllers/mno_enterprise/jpi/v1/admin/dashboard_templates_controller.rb]: make sure to add `{dashboard_type: 'template'}` in the creation parameters.

**Comment by hedudelgado** [comment]
@x4d3 , thanks for the comments. I have corrected most of them. I have not changed the aliases yet though, waiting for your input on the following.
I created the aliases to improve the readability of the code. The same method is called for the same purpose, but at different moments and different end result. One end result is update and other is create.
Shall I remove them or leave them?

**Comment by ouranos** [comment]
@cesar-tonnoir FYI
Since there might be some needs for adjustments while you guys start on the frontend, I guess it's better to wait before merging?

**Comment by ouranos** [comment]
Merging in feature branch `feature/dashboard-designer`

**Pull request merged**